src/services/exportService.ts

```diff
@@ -304,57 +361,183 @@ class ExportService {
         }
     }
 
     private async exportMotionPhoto(
         fileStream: ReadableStream<any>,
         file: File,
-        collectionPath: string
+        collectionPath: string,
+        usedFileNamesInCollection: Set<string>
     ) {
         const fileBlob = await new Response(fileStream).blob();
         const originalName = fileNameWithoutExtension(file.metadata.title);
         const motionPhoto = await decodeMotionPhoto(fileBlob, originalName);
-
         const imageStream = generateStreamFromArrayBuffer(motionPhoto.image);
-        const imageUID = `${file.id}_${motionPhoto.imageNameTitle}`;
-        this.saveMediaFile(collectionPath, imageUID, imageStream);
-        this.saveMetadataFile(collectionPath, imageUID, file.metadata);
+        const imageSaveName = getUniqueFileSaveName(
+            motionPhoto.imageNameTitle,
+            usedFileNamesInCollection
+        );
+        this.saveMediaFile(collectionPath, imageSaveName, imageStream);
+        this.saveMetadataFile(collectionPath, imageSaveName, file.metadata);
 
         const videoStream = generateStreamFromArrayBuffer(motionPhoto.video);
-        const videoUID = `${file.id}_${motionPhoto.videoNameTitle}`;
-        this.saveMediaFile(collectionPath, videoUID, videoStream);
-        this.saveMetadataFile(collectionPath, videoUID, file.metadata);
+        const videoSaveName = getUniqueFileSaveName(
+            motionPhoto.videoNameTitle,
+            usedFileNamesInCollection
+        );
+        this.saveMediaFile(collectionPath, videoSaveName, videoStream);
+        this.saveMetadataFile(collectionPath, videoSaveName, file.metadata);
     }
 
-    private saveMediaFile(collectionPath, uid, fileStream) {
+    private saveMediaFile(
+        collectionFolderPath: string,
+        fileSaveName: string,
+        fileStream: ReadableStream<any>
+    ) {
         this.ElectronAPIs.saveStreamToDisk(
-            `${collectionPath}/${uid}`,
+            getFileSavePath(collectionFolderPath, fileSaveName),
             fileStream
         );
     }
-    private saveMetadataFile(collectionPath, uid, metadata) {
+    private saveMetadataFile(
+        collectionFolderPath: string,
+        fileSaveName: string,
+        metadata: MetadataObject
+    ) {
         this.ElectronAPIs.saveFileToDisk(
-            `${collectionPath}/${METADATA_FOLDER_NAME}/${uid}.json`,
-            getGoogleLikeMetadataFile(uid, metadata)
+            getFileMetadataSavePath(collectionFolderPath, fileSaveName),
+            getGoogleLikeMetadataFile(fileSaveName, metadata)
         );
     }
 
-    private sanitizeName(name) {
-        return name.replaceAll('/', '_').replaceAll(' ', '_');
-    }
-
     isExportInProgress = () => {
         return this.exportInProgress !== null;
     };
+
+    private async migrateExport(
+        exportDir: string,
+        currentVersion: number,
+        collections: Collection[],
+        files: File[]
+    ) {
+        if (currentVersion === 0) {
```

Comment:
Would it make more sense to not take the current Version as input. We can fetch the currentVersion here and skip if it's already exported or something? Also, we can probably add some documentation/comment here about what are we doing for future :)

Reply:
ya we can totally avoid passing currentVersion. Its just that we fetched the export_status.json , the caller function so just used that to avoid duplicate get calls.

yes will document it..i am thinking to keep the design similar to how DB migration work and so we would apply rolling updates.

Reply:
for now will simply comment about what the migration does and would later on add the rolling migration when we have more than one migrations